app/controllers/admin/api/accounts_controller.rb

```diff
@@ -258,4 +249,21 @@ def account_plan
   def billing_params
     @billing_params ||= params.permit(:monthly_billing_enabled, :monthly_charging_enabled)
   end
+
+  private
+
+  def find_buyer_account
+    case
+    when username = params[:username]
+      buyer_users.find_by!(username: username).account
+    when user_id = params[:user_id]
+      buyer_users.find(user_id).account
+    when current_account.master? && provider_key = params[:buyer_provider_key]
```

Comment:
Why only for `current_account.master?` ?

Reply:
this should be master only functionality for now.

Reply:
noupe, it just complains about complexity.